Deploy more smart pointers in Source/WebKit/Shared/

rniwa opened the pull request:
#### cc21d6f48180ca2d26d5f5db4809001a3b28dd5c
<pre>
Deploy more smart pointers in Source/WebKit/Shared/
<a href="https://bugs.webkit.org/show_bug.cgi?id=260440">https://bugs.webkit.org/show_bug.cgi?id=260440</a>

Reviewed by Wenson Hsieh.

Deployed more smart pointers.

* Source/WebKit/Shared/APIWebArchive.mm:
(API::WebArchive::WebArchive):
* Source/WebKit/Shared/Authentication/AuthenticationManager.cpp:
(WebKit::AuthenticationManager::processChecked const): Added.
(WebKit::AuthenticationManager::AuthenticationManager):
(WebKit::AuthenticationManager::~AuthenticationManager):
(WebKit::AuthenticationManager::didReceiveAuthenticationChallenge):
(WebKit::AuthenticationManager::negotiatedLegacyTLS const):
* Source/WebKit/Shared/Authentication/AuthenticationManager.h:
* Source/WebKit/Shared/Cocoa/WKNSArray.mm:
(-[WKNSArray objectAtIndex:]):
* Source/WebKit/Shared/Cocoa/WKNSDictionary.mm:
(-[WKNSDictionary objectForKey:]):
* Source/WebKit/Shared/EditingRange.cpp:
(WebKit::EditingRange::toRange):
(WebKit::EditingRange::fromRange):
* Source/WebKit/Shared/IPCTester.cpp:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm:
(WebKit::RemoteLayerBackingStore::encode const):
(WebKit::RemoteLayerBackingStore::drawInContext):
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.cpp:
(WebKit::dump):
* Source/WebKit/Shared/UserData.cpp:
(WebKit::shouldTransform):
(WebKit::transformGraph):
(WebKit::UserData::encode):
* Source/WebKit/Shared/WebConnection.cpp:
(WebKit::WebConnection::handleMessage):
* Source/WebKit/Shared/WebContextMenuItem.cpp:
(WebKit::WebContextMenuItem::separatorItem):
* Source/WebKit/Shared/WebHitTestResultData.cpp:
(WebKit::WebHitTestResultData::WebHitTestResultData):
(WebKit::WebHitTestResultData::elementBoundingBoxInWindowCoordinates):

Canonical link: <a href="https://commits.webkit.org/267232@main">https://commits.webkit.org/267232@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ccd55a3667a705eb8c099ab072457b98279b3a89

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/15964 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16283 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/16670 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17727 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14988 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/19293 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16384 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17440 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16155 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/16630 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13612 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/18487 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/13874 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21298 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14864 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/14595 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17840 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15191 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12878 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14430 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18799 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1964 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15012 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/Shared/UserData.cpp:
```diff
@@ -87,7 +87,8 @@ static bool shouldTransform(const API::Object& object, const UserData::Transform
 
-            if (shouldTransform(*keyValuePair.value, transformer))
+            Ref value = *keyValuePair.value;
+            if (shouldTransform(value, transformer))
```
Just curious — any reason why we don't use `shouldTransform(Ref { *keyValuePair.value }, transformer)` instead?

rniwa replied:
I was just trying out a different style but will use that instead.

rniwa committed:
Deploy more smart pointers in Source/WebKit/Shared/
https://bugs.webkit.org/show_bug.cgi?id=260440

Reviewed by Wenson Hsieh.

Deployed more smart pointers.

* Source/WebKit/Shared/APIWebArchive.mm:
(API::WebArchive::WebArchive):
* Source/WebKit/Shared/Authentication/AuthenticationManager.cpp:
(WebKit::AuthenticationManager::processChecked const): Added.
(WebKit::AuthenticationManager::AuthenticationManager):
(WebKit::AuthenticationManager::~AuthenticationManager):
(WebKit::AuthenticationManager::didReceiveAuthenticationChallenge):
(WebKit::AuthenticationManager::negotiatedLegacyTLS const):
* Source/WebKit/Shared/Authentication/AuthenticationManager.h:
* Source/WebKit/Shared/Cocoa/WKNSArray.mm:
(-[WKNSArray objectAtIndex:]):
* Source/WebKit/Shared/Cocoa/WKNSDictionary.mm:
(-[WKNSDictionary objectForKey:]):
* Source/WebKit/Shared/EditingRange.cpp:
(WebKit::EditingRange::toRange):
(WebKit::EditingRange::fromRange):
* Source/WebKit/Shared/IPCTester.cpp:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm:
(WebKit::RemoteLayerBackingStore::encode const):
(WebKit::RemoteLayerBackingStore::drawInContext):
* Source/WebKit/Shared/RemoteLayerTree/RemoteScrollingCoordinatorTransaction.cpp:
(WebKit::dump):
* Source/WebKit/Shared/UserData.cpp:
(WebKit::shouldTransform):
(WebKit::transformGraph):
(WebKit::UserData::encode):
* Source/WebKit/Shared/WebConnection.cpp:
(WebKit::WebConnection::handleMessage):
* Source/WebKit/Shared/WebContextMenuItem.cpp:
(WebKit::WebContextMenuItem::separatorItem):
* Source/WebKit/Shared/WebHitTestResultData.cpp:
(WebKit::WebHitTestResultData::WebHitTestResultData):
(WebKit::WebHitTestResultData::elementBoundingBoxInWindowCoordinates):

Canonical link: https://commits.webkit.org/267232@main
